packages/terra-doc-template/src/IndexPageTemplate.jsx

```diff
@@ -0,0 +1,73 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Markdown from 'terra-markdown';
+import Button from 'terra-button';
+import SlidePanel from 'terra-slide-panel';
+
+const propTypes = {
+  example: PropTypes.element,
+  exampleSrc: PropTypes.string,
+  title: PropTypes.string,
+  description: PropTypes.string,
+  children: PropTypes.element,
+};
+
+const defaultProps = {
+  example: null,
+  exampleSrc: null,
+  title: null,
+  description: null,
+  children: null,
+};
+
+const codeIndicator = '```';
+
+class IndexExampleTemplate extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = { panelIsOpen: false };
+    this.handlePanelToggle = this.handlePanelToggle.bind(this);
+  }
+
+  handlePanelToggle() {
+    this.setState({ panelIsOpen: !this.state.panelIsOpen });
+  }
+
+  render() {
+    const { title, example, exampleSrc, children, description, ...customProps } = this.props;
+
+    const indexExample = (
+      <div>
+        <Button text="Show Source Code" onClick={this.handlePanelToggle} data-terra-example-hide />
```

Comment:
Lets update these data-attributes to be `data-terra-example-template-hide` and `data-terra-example-template-show`